instrumentation/wildfly/src/main/java/com/splunk/opentelemetry/middleware/WildFlyAttributesInstrumentationModule.java

```diff
@@ -69,12 +69,12 @@ public void transform(TypeTransformer typeTransformer) {
   public static class MiddlewareInitializedAdvice {
     @Advice.OnMethodEnter
     public static void onEnter() {
-      CallDepthThreadLocalMap.incrementCallDepth(ProductConfig.class);
+      CallDepth.forClass(ProductConfig.class).getAndIncrement();
```

Comment:
Should we pass `CallDepth` between enter/exit methods using a local variable?

Reply:
Sure. This PR or a follow up?

Reply:
I will do a separate PR after this one.